source/elements/oneDAL/source/data_management/data_sources/index.rst

```diff
@@ -0,0 +1,157 @@
+  provides the ability to load data by batches and extracts it directly into the
+  device's local memory. Therefore, a data source enables complex data analytics
+  scenarios, such as :capterm:`online computations <Online mode>`.
+
+- To filter and normalize :capterm:`feature` values that are being extracted.
+
+- To recover missing :capterm:`feature` values.
+
+- To detect :capterm:`outliers <outlier>` and recover the abnormal data.
+
+- To transform datasets into numerical representation. Data source shall
+  automatically transform non-numeric :capterm:`categorical <categorical
+  feature>` and :capterm:`continuous <continuous feature>` data values into one
+  of the numeric :capterm:`data formats <data format>`.
+
+CSV data source
+-----------------
+Class ``csv::data_source`` is an implementation of a data source type
+for which the following is true:
+
+::
+
+::
+
+   class data_source {
+   public:
+      csv_data_source(const char *file_name);
+      csv_data_source(const std::string &file_name);
+
+      char get_delimiter() const;
+      bool get_parse_header() const;
+      std::string get_file_name() const;
+
+      csv_data_source& set_delimiter(char);
+      csv_data_source& set_parse_header(bool);
+      csv_data_source& set_file_name(const char *);
+      csv_data_source& set_file_name(const std::string &);
+   };
+
+.. namespace:: onedal::csv
+.. class:: data_source
+
+   .. function:: data_source(const char *file_name)
+
+      Creates new instance of csv data source with the given :expr:`file_name`, the
+      other attributes get default values.
+
+   .. function:: data_source(const std::string &file_name)
+
+      Creates new instance of csv data source with the given :expr:`file_name`, the
+      other attributes get default values.
+
+   .. member:: char delimiter = ','
+
+      Symbol that represents delimiter between continuous feature columns in the input file.
+
+      Getter & Setter
+         | ``char get_delimter() const``
+         | ``data_source& set_delimiter(char)``
+
+   .. member:: bool parse_header = false
+
+      Flag that indicates whether or not first line in the input file should be processed
+      as a header line and should not be included in the dataset.
+
+      Getter & Setter
+         | ``bool get_parse_header() const``
+         | ``data_source& set_parse_header(bool)``
+
+   .. member:: std::string file_name = ""
+
+      String which contains name of the file with the dataset for read.
+
+      Getter & Setter
+         | ``std::string get_filename() const``
+         | ``data_source& set_file_name(const char *)``
+         | ``data_source& set_file_name(const std::string &)``
+
+-------------
+Usage example
+-------------
+::
+
+   const auto data_source = onedal::csv::data_source("data.csv")
+      .set_delimiter(',')
+
+   const auto table = ondedal::read<onedal::table>(data_source);
+
+
+Reading :expr:`onedal::read<onedal::table>(...)`
+-----------------------------------
+Args
```

Comment:
it seems w/o description of csv in the beginning this 'arg' section would raise the questions what it means for the implementer

Reply:
Actually, args class will be empty for this release, so we may consider excluding this section for now.